### PR TITLE
Add CockroachDB support.

### PR DESCRIPTION
**Pull request by turnkey-commerce**
This PR implements the support for [CockroachDB](https://github.com/cockroachdb/cockroach) in xo (Issue #119).

## Testing
Added **examples/booktest/cockroachdb** for testing. To test it you must [install CockroachDB](https://www.cockroachlabs.com/docs/stable/install-cockroachdb.html) then use the script **contrib/crcreate.sh** to setup the CockroachDB before running the test.  Then run the **examples/booktest/gen.sh** to confirm the results.  I also confirmed the examples/booktest/postgres is working as before.

## Implementation
The changes to get xo to work with CockroachDB involved minor modification to the existing Postgres logic - CockroachDB is wire-compatible with Postgres and uses the same Go driver.  The modifications are designed to not affect existing Postgres logic:

1.  Check for support of pg_get_function_result and if it  is not present do not try to invoke it (was previously crashing in CockroachDB). Stored procedures and functions are not yet supported in CockroachDB.
2. Support the way the logic for auto-increment primary keys are handled in CockroachDB, by querying for tables with a column with a constraint that is named 'primary' and the column in question has a default value of 'unique_rowid()'.

The README has been updated to show the new capabilities and to also point out differences in getting the results with a detail about how database names work in CockroachDB.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (andreis) [inline .vscode/launch.json]: Perhaps this file shouldn't be included?

File: examples/booktest/.gitignore [hunk 2]
```diff
@@ -1,2 +1,2 @@
 */skip
-/booktest-*
+/booktest-*
```
Comment (andreis) [inline examples/booktest/.gitignore]: Not necessary

File: .vscode/launch.json (hunk 1 above)
Comment (turnkey-commerce) [reply .vscode/launch.json]: Thanks for the review, it's definitely not necessary although it is helpful to have around to debug in VSCode.  However not everybody uses it of course so can remove it.

File: examples/booktest/.gitignore (hunk 2 above)
Comment (turnkey-commerce) [reply examples/booktest/.gitignore]: Thanks, will fix.

**Comment by andreis** [comment]
Much appreciated James, and thanks for your work!

We'll give back to the community by publishing our xo fork (whenever we'll have time for that) - we've customized it heavily and create our GraphQL APIs with it.

**Comment by kennysh** [comment]
I get an error: "Scan error on column index 1: converting driver.Value type <nil> ("<nil>") to a int: invalid syntax"
File: model/index.xo.go
line: 50

**Comment by yml** [comment]
Hello,

I am not sure how to read the fact that this PR is almost a year old. Can we now use XO and directly using the fact that it is postgres compatible ? 

Is the changes in this PR still relevant ? What are the blockers to get this merged ?
Thank you

**Comment by turnkey-commerce** [comment]
@yml, as far as I know it is still relevant for the reasons mentioned in the PR documentation, especially the second point about the primary keys as they are very different in CockroachDB.  So I don't think the current XO will work with it directly in its present state.  I think the only blocker is that the maintainer has gone on hiatus, I'll try to contact him directly about including it.
Cheers,
James

**Comment by turnkey-commerce** [comment]
@kenny-yu could you please include your create statements to create the CockroachDB that had the error?  I ran the examples/booktest/cockroachdb with the PR on the latest CRDB version and I didn't reproduce the error.  One other note because of a change in xo to fix the proper package paths I had to do a global replace in the code of "knq/xo" to "xo/xo".

@andreis if you are seeing a similar error also please send me your create statements.

**Comment by turnkey-commerce** [comment]
@andreis no worries, thanks for the reply.

**Comment by turnkey-commerce** [comment]
FYI - this PR has been incorporated into the fork at https://github.com/turnkey-commerce/gendal.

**Pull request closed without merging**